mslib/mswms/wms.py

```diff
@@ -548,9 +612,83 @@ def produce_plot(self, query, mode):
+                    if init_time is None:
+                        return self.create_service_exception(
+                            code="MissingDimensionValue",
+                            text="INIT_TIME not specified (use the DIM_INIT_TIME keyword)",
+                            version=version)
+                    if valid_time is None:
+                        return self.create_service_exception(code="MissingDimensionValue",
+                                                             text="TIME not specified",
+                                                             version=version)
+
+                # Bounding box (num interp. points, num labels).
+                try:
+                    bbox = [float(v) for v in query.get("BBOX", "101,10").split(",")]
+                except ValueError:
+                    return self.create_service_exception(text=f"Invalid BBOX: {query.get('BBOX')}", version=version)
+
+                plot_driver = self.lsec_drivers[dataset]
+                try:
+                    plot_driver.set_plot_parameters(plot_object=self.lsec_layer_registry[dataset][layer],
+                                                    lsec_path=path,
+                                                    lsec_numpoints=bbox[0],
+                                                    lsec_path_connection="greatcircle",
+                                                    lsec_numlabels=bbox[1],
+                                                    init_time=init_time,
+                                                    valid_time=valid_time,
+                                                    style=style,
+                                                    bbox=bbox,
+                                                    figsize=figsize,
+                                                    noframe=False,
+                                                    transparent=transparent,
+                                                    color=color,
+                                                    return_format=return_format)
+                    images.append(plot_driver.plot())
+                except (IOError, ValueError) as ex:
+                    logging.error("ERROR: %s %s", type(ex), ex)
+                    msg = "The data corresponding to your request is not available. Please check the " \
+                          "times and/or path you have specified.\n\n" \
+                          f"Error message: {ex}.\n" \
+                          "Hint: Check used waypoints."
+                    return self.create_service_exception(text=msg, version=version)
+
         # 4) Return the produced image.
         # =============================
-        return squash_multiple_images(images), return_format
+        if len(layers) > 1:
+            if "image" in return_format:
+                return squash_multiple_images(images), return_format
+            elif "xml" in return_format:
```

Comment:
I dislike having branches we might run into in an error case.
I would add something like assert False or raise RunTimeError in an additional "else:" branch, which *should* never be triggered. But failing spectacular is often better than doing the same silently...

Reply:
Done